### PR TITLE
[Leduong]:controller-middleware

### PR DESCRIPTION
**Pull request by leduong290300**
+) Add form login && register
+) Create controller && resource controller
+) Validate register form
+) Auth user

https://user-images.githubusercontent.com/80597229/133897406-46e70219-6679-43aa-8d6a-873bfd8f6c09.mp4

### CODE REVIEW

File: Le_duong/Laravel/eCommerce/app/Http/Controllers/LoginController.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//        'email' => ['required','max:100','email'],
+//        'password' => ['required','between:8,32'],
+//      ]);
+
+      $data = $request->only('email', 'password');
+      if(Auth::attempt($data)) {
+        return redirect()->route('dashboard');
+      } else {
+        return redirect()->route('login');
+      }
```
Comment (giangnv-0040) [inline Le_duong/Laravel/eCommerce/app/Http/Controllers/LoginController.php:28]:
```suggestion
      if (Auth::attempt($data)) {
          return redirect()->route('dashboard');
      }
      
      return redirect()->route('login');
```

File: Le_duong/Laravel/eCommerce/app/Http/Controllers/RegisterController.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    //Load form register form
+    public function create()
+    {
+        return view('pages.register');
+    }
+
+    //Save data to db
+    public function register(Request $request)
+    {
+    $first_name = $request->input('first_name');
```
Comment (giangnv-0040) [inline Le_duong/Laravel/eCommerce/app/Http/Controllers/RegisterController.php:19]: indent e ơi

File: Le_duong/Laravel/eCommerce/app/Http/Middleware/Admin/Auth.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        $currentUser = auth()->user();
+        if($currentUser->email == 'admin@gmail.com')
+        {
```
Comment (giangnv-0040) [inline Le_duong/Laravel/eCommerce/app/Http/Middleware/Admin/Auth.php:21]:
```suggestion
        if ($currentUser->email == 'admin@gmail.com') {
```

File: Le_duong/Laravel/eCommerce/app/Http/Controllers/RegisterController.php (hunk 2 above)
Comment (leduong290300) [reply Le_duong/Laravel/eCommerce/app/Http/Controllers/RegisterController.php:19]: ok a xong e xửa a

**Pull request merged**